packages/api/src/error-handler.js

```diff
@@ -6,18 +6,23 @@ import { HTTPError, PinningServiceApiError } from './errors.js'
  * @param {Error & {status?: number;code?: string;reason?: string; details?: string; }} err
  * @param {import('./env').Env} env
  */
-export function errorHandler (err, { log }) {
+export function errorHandler (err, { log }, request) {
   console.error(err.stack)
 
   let status = err.status || 500
   if (status >= 500) {
     log.error(err)
   }
 
-  if (err instanceof PinningServiceApiError) {
+  // As the pinning service requires a certain specification
+  // Always return the nested error object for any pinning service endpoints
+  const requestUrl = new URL(request.url)
+  if (err instanceof PinningServiceApiError || (requestUrl && /^\/pins/.test(requestUrl.pathname))) {
     const error = {
-      reason: err.reason,
-      details: err.details
+      error: {
```

Comment:
nit. Please ignore this if you don't think is a good idea.

I was wondering if we coud add a `static` method on `PinningServiceApiError` to have an API like this:
````
const pErr = PinningServiceApiError.fromHTTPError(err)
```

I think it'd be neat to also have the error classes be responsible to encode themselves in Json errors, but that's defo outside of this ticket.

Reply:
Yes that might be a neat idea actually if it ever comes up again. @mbommerez is it worth making a note of this somewhere?